yb-voyager/src/query/sqltransformer/sqlfiletransformer.go

```diff
@@ -70,32 +82,24 @@ func (t *IndexFileTransformer) Transform(file string) (string, error) {
 
 	parseTree, err = queryparser.ParseSqlFile(file)
 	if err != nil {
-		//In case of PG we should return error but for other SourceDBTypes we should return nil as the parser can fail
-		errMsg := fmt.Errorf("failed to parse %s: %w", file, err)
-		if t.sourceDBType != constants.POSTGRESQL {
-			log.Infof("skipping error while parsing %s: %v", file, err)
-			errMsg = nil
-		}
-		return "", errMsg
+		return "", fmt.Errorf("failed to parse %s: %w", file, err)
 	}
 
 	transformer := NewTransformer()
+	if !t.skipPerformanceOptimizations {
 
-	var removedSqlStmts []string
-	if !t.skipPerformanceOptimizations && t.sourceDBType == constants.POSTGRESQL {
-		parseTree.Stmts, removedSqlStmts, t.RemovedRedundantIndexes, err = transformer.RemoveRedundantIndexes(parseTree.Stmts, t.redundantIndexesToRemove)
+		//remove redundant indexes
+		var removedIndexToStmtMap *utils.StructMap[*sqlname.ObjectNameQualifiedWithTableName, *pg_query.RawStmt]
+		parseTree.Stmts, removedIndexToStmtMap, err = transformer.RemoveRedundantIndexes(parseTree.Stmts, t.redundantIndexesToExistingIndexToRemove)
 		if err != nil {
-			return "", fmt.Errorf("failed to remove redundant indexes: %w", err)
+			return "", fmt.Errorf("failed to remove redundant indexes: %w\n%s", err, SUGGESTION_TO_USE_SKIP_PERF_OPTIMIZATIONS_FLAG)
 		}
-		if len(removedSqlStmts) > 0 {
-			// Write the removed indexes to a file
-			t.RedundantIndexesFileName = filepath.Join(filepath.Dir(file), "redundant_indexes.sql")
-			redundantIndexContent := strings.Join(removedSqlStmts, "\n\n")
-			err = os.WriteFile(t.RedundantIndexesFileName, []byte(redundantIndexContent), 0644)
-			if err != nil {
-				return "", fmt.Errorf("failed to write redundant indexes file: %w", err)
-			}
+		t.RedundantIndexesFileName = filepath.Join(filepath.Dir(file), REMOVED_REDUNDANT_INDEXES_FILE_NAME)
+		err = t.writeRemovedRedundantIndexesToFile(removedIndexToStmtMap)
+		if err != nil {
+			return "", fmt.Errorf("failed to write removed redundant indexes to file: %w\n%s", err, SUGGESTION_TO_USE_SKIP_PERF_OPTIMIZATIONS_FLAG)
```

Comment:
Ideally, this suggestion should be provided for any error that comes out for any of the operations involved (i.e. parse, transform, deparse, write, etc). Let's think about this later, it's okay for now.